### PR TITLE
Compute sna/atom on fixed number of neighbors and compute slcsa/atom (Supervised Learning Crystal Structure Analysis tool) 

### PR DESCRIPTION
**Pull request by lafourcadep**
**Summary**

This pull request concerns the addition of an option to the compute sna/atom style to be able to compute the bispectrum over a fixed target number of neighbors as well as a new compute slcsa/atom style that allows to take the bispectrum as input, perform a dimension reduction followed by a logistic regression classification. These two tools allow for a on-th-efly crystal structure classification in LAMMPS simulations. 
Related publication : 
https://www.sciencedirect.com/science/article/pii/S0927025623005281

After discussion with Aidan and Steve, I added the option to the existing compute sna/atom style instead of duplicating code. The only thing that changes is that the neighborhood contains a fixed number of neighbors instead of the full neighbor list contained in the interaction sphere based on the cutoff radius.

**Related Issue(s)**

No related issues.

**Author(s)**

Paul Lafourcade 
work e-mail : paul.lafourcade@cea.fr
long-lived e-mail address : lafourcadepaul16@gmail.com
1) CEA DAM DIF, 91297 Arpajon, France
2) Université Paris-Saclay, LMCE, 91680 Bruyères-le-Châtel, France

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

Changes were made to the compute_sna_atom.cpp/h files by adding optional keywords to the compute style.
The existing compute style is still working and I included 

**Implementation Notes**

Implementation is based on other compute_style and follows the LAMMPS guidelines.

**Post Submission Checklist**

- [X] The feature or features in this pull request is complete
- [X] Licensing information is complete
- [X] Corresponding author information is complete
- [X] The source code follows the LAMMPS formatting guidelines
- [X] Suitable new documentation files and/or updates to the existing docs are included
- [X] The added/updated documentation is integrated and tested with the documentation build system
- [X] The feature has been verified to work with the conventional build system
- [X] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [X] One or more example input decks are included

**Further Information, Files, and Links**

https://www.sciencedirect.com/science/article/pii/S0927025623005281

### CODE REVIEW

**Review by athomps — COMMENTED** [review]
This looks good. However, `examples/sna_nnn_slcsa` is not sufficiently important to go in the top level of `examples/` dir. I suggest putting it inside `examples/snap` or else inside `examples/PACKAGES`

**Comment by akohlmey** [comment]
> This looks good. However, `examples/sna_nnn_slcsa` is not sufficiently important to go in the top level of `examples/` dir. I suggest putting it inside `examples/snap` or else inside `examples/PACKAGES`

Updated the example to be more useful for testing, shorter, more in line with other example conventions, added logs and moved the folder.

File: examples/PACKAGES/sna_nnn_slcsa/dir.slcsa/lr_bias.dat [hunk 1]
```diff
@@ -0,0 +1 @@
+-6.32012657 5.62127377 1.19871662 -0.49986382
```
Comment (stanmoore1) [inline examples/PACKAGES/sna_nnn_slcsa/dir.slcsa/lr_bias.dat]: Need to add a newline at end of file?

**Review by stanmoore1 — APPROVED** [review]
One comment, otherwise looks good

File: examples/PACKAGES/sna_nnn_slcsa/dir.slcsa/lr_bias.dat (hunk 1 above)
Comment (akohlmey) [reply examples/PACKAGES/sna_nnn_slcsa/dir.slcsa/lr_bias.dat]: It worked *as is*. I tweaked the input and ran it to create logs (and have the compute output to thermo data by outputting min/max of compute per-atom properties).

**Pull request merged**